### PR TITLE
[AF-488] Support hash params for origin and app guid

### PR DESCRIPTION
**Pull request by adammw**
:v:

/cc @zendesk/vegemite

### Description
Support hash params for origin and app guid

### References
* JIRA: https://zendesk.atlassian.net/browse/AF-488

### Risks
* medium: handshake breaks, causing all v2 api calls to fail/error out

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -25,12 +25,15 @@ var Client    = require('client'),
 /// });
 /// ```
 ZAFClient.init = function(callback) {
-  var params = Utils.queryParameters(),
+  var queryParams = Utils.queryParameters(),
+      hashParams = Utils.queryParameters(( document.location.hash || '' ).slice(1)),
```
Comment (sandlerr) [inline lib/index.js]: redundant `document`?

Comment (adammw) [inline lib/index.js]: possibly, had copied from the source of `Utils.queryParameters` which did the same thing... doesn't hurt does it?

File: lib/index.js [hunk 2]
```diff
@@ -25,12 +25,15 @@ var Client    = require('client'),
...
+      origin = queryParams.origin || hashParams.origin,
+      app_guid = queryParams.app_guid || hashParams.app_guid,
```
Comment (danielbreves) [inline lib/index.js]: `appGuid`?

Comment (adammw) [inline lib/index.js]: this is the the query parameter, which uses snake case...

Comment (danielbreves) [inline lib/index.js]: I know, we need to leave the parameter as is, but we can rename the `var`, no?

**Pull request merged**